src/scripts/main.js

```diff
@@ -1,3 +1,32 @@
 'use strict';
 
-// write code here
+// отримуємо коллекцію усіх елементів списку
+const headline = document.querySelectorAll('li');
+
+function createSpan(el) {
+  const spanEl = document.createElement('span');
+  const textNode = el.firstChild;
+
+  el.firstChild.parentNode.insertBefore(spanEl, textNode);
+  spanEl.appendChild(textNode);
+}
+
+for (const el of headline) {
+  if (el.children.length > 0) {
+    createSpan(el);
+  }
+}
+
+const spanArray = [...document.querySelectorAll('span')];
+
+for (const li of spanArray) {
+  li.addEventListener('click', (e) => {
+    const children = li.nextElementSibling;
+
+    if (children.style.display === '') {
+      children.style.display = 'none';
+    } else {
+      children.style.display = '';
```

Comment:
The logic here assumes that `children.style.display` is initially an empty string. However, if `children` is `null`, this will throw an error. Ensure that `children` is not `null` before attempting to access or modify its `style` property.